pkg/manifest/kustomization.go

```diff
@@ -0,0 +1,197 @@
+package manifest
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+	appsv1 "k8s.io/api/apps/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/kustomize/api/image"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/resource"
+	kustypes "sigs.k8s.io/kustomize/api/types"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+
+	"github.com/xenitab/gitops-promotion/pkg/git"
+)
+
+const (
+	kustomizationFile = "kustomization.yaml"
+	featureLabel      = "gitops-promotion.xenit.io/feature"
+)
+
+// DuplicateApplication duplicates the application manifests based on the label selector.
+// It assumes that the fs is a base fs in the repository directory.
+// nolint:gocognit,gocritic // ignore
+func DuplicateApplication(fs afero.Fs, state git.PRState, labelSelector map[string]string) error {
+	envPath := filepath.Join(state.Group, state.Env)
+
+	// Write feature app manifests
+	resources, err := manfifestsMatchingSelector(fs, envPath, labelSelector)
+	if err != nil {
```

Comment:
Will this trigger with zero resources? Is it ok to have no resources in your feature?